TorchSharp/TypeGeneration.tt

```diff
@@ -1271,6 +1466,130 @@ foreach (var type in new [] {
             TH<#=tname#>Tensor_indexCopy (handle, dim, index.handle, src.handle);
         }
 
+        [DllImport ("caffe2")]
+        extern static void TH<#=tname#>Tensor_indexAdd (HType tensor, int dim, LongTensor.HType index, HType src);
+        
+        /// <summary>
+        ///   Adds the elements of tensor into the original tensor by selecting the indices in the order 
+        ///   given in index. The shape of tensor must exactly match the elements indexed or an error will be thrown.
+        /// </summary>
+        /// <param name="dim">Dimension to select for the add</param>
+        /// <param name="index">Entries to copy</param>
+        /// <param name="src">Tensor to copy the data from.</param>
+        public void IndexAdd (int dim, LongTensor index, <#=tname#>Tensor src)
+        {
+            if (src == null)
+                throw new ArgumentNullException (nameof (src));
+            if (index == null)
+                throw new ArgumentNullException (nameof (index));
+            TH<#=tname#>Tensor_indexAdd (handle, dim, index.handle, src.handle);
+        }
+
+        [DllImport ("caffe2")]
+        extern static void TH<#=tname#>Tensor_indexFill (HType tensor, int dim, LongTensor.HType index, <#=ttype#> value);
+        
+        /// <summary>
+        ///   Uses the given value to overwrite the original tensor by selecting the indices in the order 
+        ///   given in index. The shape of tensor must exactly match the elements indexed or an error will be thrown.
+        /// </summary>
+        /// <param name="dim">Dimension to select for the fill</param>
+        /// <param name="index">Entries to copy</param>
+        /// <param name="value">The value to write.</param>
+        public void IndexFill (int dim, LongTensor index, <#=ttype#> value)
+        {
+            if (index == null)
+                throw new ArgumentNullException (nameof (index));
+            TH<#=tname#>Tensor_indexFill (handle, dim, index.handle, value);
+        }
+
+        [DllImport ("caffe2")]
+        extern static void TH<#=tname#>Tensor_take (HType self, HType src, LongTensor.HType index);
+
+        /// <summary>
+        ///   Take
+        /// </summary>
+        public void Take (<#=tname#>Tensor src, LongTensor index)
+        {
+            if (src == null)
+                throw new ArgumentNullException (nameof (src));
+            if (index == null)
+                throw new ArgumentNullException (nameof (index));
+            TH<#=tname#>Tensor_take (handle, src.handle, index.handle);
+        }
+
+        [DllImport ("caffe2")]
+        extern static void TH<#=tname#>Tensor_put (HType self, LongTensor.HType index, HType src, int accumulate);
+
+        /// <summary>
+        ///   Put
```

Comment:
Doc please. Here and the successive ops

Reply:
Problem is, I don't actually know precisely what these are doing -- for now, I've been focusing on coverage.

Reply:
Yes, that makes sense. I'll create it.

Reply:
See #35